### PR TITLE
Connection management ui

### PR DESCRIPTION
**Pull request by agatsoh**
Makes a dialog for user to Join a token network through the raiden webui

### CODE REVIEW

File: raiden/raidenwebui/src/app/components/token-network/token-network.component.ts [hunk 1]
```diff
@@ -44,17 +53,30 @@ export class TokenNetworkComponent implements OnInit {
...
-                this.sharedService.msg({
-                    severity: 'warn',
-                    summary: 'Warining message',
-                    detail: 'You already participate in this token network'
-                });
-                return;
-            }
-        }
-        this.displayDialog = true;
+        this.displayJoinDialog = true;
```
Comment (andrevmatos) [inline raiden/raidenwebui/src/app/components/token-network/token-network.component.ts]: I'd recommend a `this.funds.reset()` before this line too, to asure input field is empty on dialog show.

File: raiden/raidenwebui/src/app/components/token-network/token-network.component.ts [hunk 2]
```diff
@@ -44,17 +53,30 @@ export class TokenNetworkComponent implements OnInit {
...
+                    } else if (response.status === 500) {
+                        this.sharedService.msg({
+                            severity: 'error',
+                            summary: 'Server Error',
+                            detail: 'Server has encountered Internal Error'
+                        })
+                    }
+                }
+          );
+          this.displayJoinDialog = false;
```
Comment (andrevmatos) [inline raiden/raidenwebui/src/app/components/token-network/token-network.component.ts]: And, maybe, resetting after dialog is closed too.

File: raiden/raidenwebui/src/app/services/raiden.service.ts [hunk 3]
```diff
@@ -168,6 +168,18 @@ export class RaidenService {
...
     }
 
+    public connectTokenNetwork(funds: number, tokenAddress: string): Observable<any> {
+        const data = {
+            'funds': funds
+        }
+        const headers = new Headers({ 'Content-Type': 'application/json' });
+        const options = new RequestOptions({ headers: headers });
+        return this.http.put(`${this.config.apiCall}/connection/${tokenAddress}`,
+        JSON.stringify(data), options).map((response) => {
```
Comment (andrevmatos) [inline raiden/raidenwebui/src/app/services/raiden.service.ts]: I think you can indent these lines under put, to visually indicate it is below `put`, and `.catch` on its own line too.

File: raiden/raidenwebui/src/app/components/token-network/token-network.component.ts (hunk 1 above)
Comment (agatsoh) [inline raiden/raidenwebui/src/app/components/token-network/token-network.component.ts]: Yes will make this change thank you.

File: raiden/raidenwebui/src/app/components/token-network/token-network.component.ts (hunk 2 above)
Comment (agatsoh) [inline raiden/raidenwebui/src/app/components/token-network/token-network.component.ts]: Noted

File: raiden/raidenwebui/src/app/services/raiden.service.ts (hunk 3 above)
Comment (agatsoh) [inline raiden/raidenwebui/src/app/services/raiden.service.ts]: Noted

**Comment by andrevmatos** [comment]
Sorry for the additional merge, it was through github' interface, so I didn't noticed until final merge

**Comment by LefterisJP** [comment]
@andrevmatos no worries. You can click on the arrow next to the "merge" and choose "Rebase and merge" to avoid it.

**Pull request merged**